src/functional/ffront/decorator.py

```diff
@@ -260,6 +260,36 @@ def format_itir(
             offset_provider=offset_provider,
         )
 
+    def _refactor_args_kwargs(self, *args, **kwargs) -> tuple[tuple, dict]:
+        past_params = self.past_node.params
+
+        new_args_ls = [None] * len(past_params)
+        wrong_args = [False] * len(past_params)
+        kwargs_count = 0
+        if len(past_params) < len(args) + len(kwargs):
+            raise GTTypeError(
+                "Number of parameters in function call exceeds number in function definition."
+            )
+        for param_i, param in enumerate(past_params):
+            if param.id in kwargs:
+                new_args_ls[param_i] = kwargs[param.id]
+                kwargs.pop(param.id)
+                kwargs_count += 1
+            elif len(args) > param_i - kwargs_count:
+                new_args_ls[param_i] = args[param_i - kwargs_count]
```

Comment:
```suggestion
            elif param_i+1 >= len(args):
                new_args_ls[param_i] = args[param_i]
```
The idea behind having `new_args_ls` was to avoid avoid exactly this kind of index computations for the access, which is not correct here. Surprisingly the tests don't fail, not exactly sure why. If you consider such a case:
```python
def func(a, b, c, d):
    ...
func(a, b, c, d=1)
```
You execute the following:
```python
new_args_ls[0] = args[0 - 1]
new_args_ls[1] = args[1 - 1]
```

Reply:
this is not correct, we can discuss on Monday